server/test/unit/server/webservices/views/test_util.py

```diff
@@ -54,3 +54,16 @@ def test_generate_json_response_with_pulp_encoder(self, mock_json):
         test_content = {'foo': 'bar'}
         util.generate_json_response_with_pulp_encoder(test_content)
         mock_json.dumps.assert_called_once_with(test_content, default=pulp_json_encoder)
+
+    @mock.patch('pulp.server.webservices.views.util.iri_to_uri')
+    def test_generate_redirect_response(self, mock_iri_to_uri):
+        """
+        Test HttpResponseRedirect.
+        """
+        test_content = {'foo': 'bar'}
+        href = '/some/url/'
+        response = HttpResponse(content=test_content)
+        redirect_response = util.generate_redirect_response(response, href)
+        self.assertEqual(redirect_response.status_code, 201)
+        self.assertEqual(redirect_response.reason_phrase, 'CREATED')
+        mock_iri_to_uri.assert_called_once_with(href)
```

Comment:
Try adding an assert statement to verify that the Location header is set to the return value of mock_iri_to_uri. From my own debugging I think the header is stored at redirect_response._headers['location']. I recommend letting mock auto-create the return_value which you can access with mock_iri_to_uri.return_value.